objcgen/operatoroverloads.cs

```diff
@@ -64,16 +72,33 @@ static OperatorOverloads ()
 					matches.AddValue (info, method);			
 			}
 
-			foreach (var match in matches.Where (x => x.Value.Count > 1)) {
-				// If we have a friendly method, ignore op variants and vice versa
-				MethodInfo friendlyMethod = match.Value.FirstOrDefault (x => !x.Name.StartsWith ("op_", StringComparison.Ordinal));
-				if (friendlyMethod != null) {
-					foreach (var opMethod in match.Value.Where (x => x.Name.StartsWith ("op_", StringComparison.Ordinal)))
-						yield return opMethod;
-				} else {
-					yield return friendlyMethod;
+			foreach (var match in matches) {
+				if (match.Key.CustomAction != null) {
+					foreach (MethodInfo ignore in match.Key.CustomAction (match.Key, match.Value, equals))
+						yield return ignore;
+					continue;
+				}
+
+				if (match.Value.Count > 1) {
+					// If we have a friendly method, ignore op variants and vice versa
+					MethodInfo friendlyMethod = match.Value.FirstOrDefault (x => !x.Name.StartsWith ("op_", StringComparison.Ordinal));
+					if (friendlyMethod != null) {
+						foreach (var opMethod in match.Value.Where (x => x.Name.StartsWith ("op_", StringComparison.Ordinal)))
+							yield return opMethod;
+					}
+					else {
+						yield return friendlyMethod;
+					}
 				}
 			}
 		}
+
+		static IEnumerable<MethodInfo> HandleEquals (OperatorInfo info, IEnumerable<MethodInfo> methods, IEnumerable<MethodInfo> equals)
```

Comment:
I did it this way with customization instead of hard coding since I expect more tweaks in future.